File: Print this pattern!/Print_this_pattern!.cpp
class Solution
{
public:
    void printPattern(int n)
    {   
        
        for(int i =n; i>0; i--) {
            string temp = "";
            for(int j =n; j>0; j--) {
                if(j>=i) temp += to_string(j);
                else temp += to_string(i);
            }
            string tt = temp;
            reverse(temp.begin(),temp.end());
            cout<<tt.substr(0,tt.size()-1)<<temp<<endl;
        }
        
        for(int i = 2; i<=n; i++) {
            string temp = "";
            for(int j =n; j>0; j--) {
                if(j>=i) temp += to_string(j);
                else temp += to_string(i);
            }
            string tt = temp;
            reverse(temp.begin(),temp.end());
            cout<<tt.substr(0,tt.size()-1)<<temp<<endl;
        }
    }
};

File: Print this pattern!/Print_this_pattern!.java
class Solution{
    void printPattern(int n){
        for (int i = n; i > 0; i--) {
            StringBuilder temp = new StringBuilder();
            for (int j = n; j > 0; j--) {
                if (j >= i) temp.append(j);
                else temp.append(i);
            }
            String tt = temp.toString();
            String reversed = new StringBuilder(temp).reverse().toString();
            System.out.println(tt.substring(0, tt.length() - 1) + reversed);
        }
        
        for (int i = 2; i <= n; i++) {
            StringBuilder temp = new StringBuilder();
            for (int j = n; j > 0; j--) {
                if (j >= i) temp.append(j);
                else temp.append(i);
            }
            String tt = temp.toString();
            String reversed = new StringBuilder(temp).reverse().toString();
            System.out.println(tt.substring(0, tt.length() - 1) + reversed);
        }
    }
}